File: solutions/algomaster/algorithms/sorting/sortColors.py
class Solution:
    def sortColors(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        mapping = {}
        for num in nums:
            if num not in mapping:
                mapping[num] = 1
            else:
                mapping[num] += 1
        
        idx = 0

        for i in range(3):
            freq = mapping.get(i, 0)
            nums[idx: idx + freq] = [i] * freq
            idx += freq

        

File: solutions/grind75/week5/sortColors.java
class Solution {
    public void sortColors(int[] nums) {
        HashMap<Integer, Integer> mapping = new HashMap<>();

        for (int n : nums) {
            mapping.put(n, mapping.getOrDefault(n, 0) + 1);
        }

        int id = 0;
        for (int i = 0; i < 3; i++) {
            int freq = mapping.getOrDefault(i, 0);
            Arrays.fill(nums, id, id + freq, i);
            id += freq;
        }
    }
}